### PR TITLE
Code fixes to address PVS report

### PR DESCRIPTION
**Pull request by BjossiAlfreds**
Rogue version of https://github.com/yquake2/xatrix/pull/32 that addresses bug https://github.com/yquake2/rogue/issues/19. Some rogue-specific stuff came up but nothing really different from the xatrix stuff.

### CODE REVIEW

**Comment by DanielGibson** [comment]
A quick reminder, in case you plan to do the same for yq2 itself: I already started that, see https://github.com/yquake2/yquake2/tree/pvs-studio-fixes
Just to prevent duplicate work :)

**Comment by BjossiAlfreds** [comment]
Thanks for the heads-up! I hope you had not started on rogue or xatrix locally. In retrospect I should have asked first if anyone was planning to work on these, I figured since no one had said anything it was still open, and in turn made exactly the mistake of being silent. :-)

**Comment by DanielGibson** [comment]
No worries, I haven't done anything on rogue and xatrix (and I don't have any local changes for yq2 itself either, that's in the latest commit which is just a bunch of FIXMEs).
FWIW, I'm currently not working on that either (too busy with work etc)

**Comment by Yamagi** [comment]
I found a nice, ling standig bug during the review. :) I've added some inline comments.

**Comment by BjossiAlfreds** [comment]
Hmm I wasn't notified of any comments, could you tell me which files you commented on?

**Comment by Yamagi** [comment]
Github had some problem yesterday... They showed commits in the webinterface that weren't available over 'git pull', some comments were only saved at the second or even third try and apparently they didn't send out mails. I added some comments to the code: https://github.com/yquake2/rogue/pull/33/files to `g_newweap.c` and `g_weapon.c`.

**Comment by BjossiAlfreds** [comment]
I've scrolled through them several times and can not see anything. I've never used this feature so I don't really know what I'm looking for, is it a text bubble or some icon I need to click?

**Comment by Yamagi** [comment]
It looks like that I must hit 'Request changes' before non-members can see the review comments. That's new, the last time I used the review features they were visible to anyone. You should be able to see the comments now. Sorry, I didn't know that.

**Comment by BjossiAlfreds** [comment]
I see it now, finally. :-)

File: src/g_newweap.c [hunk 1]
```diff
@@ -1486,7 +1478,7 @@ blaster2_touch(edict_t *self, edict_t *other, cplane_t *plane, csurface_t *surf)
 	int mod;
 	int damagestat;
 
-	if (!self || !other || !plane || !surf)
```
[the remark this answers is not in the document]
Comment (BjossiAlfreds) [reply src/g_newweap.c]: I was on the fence on the !plane checks, actually. In my own code I tend to prefer maximal execution - make the function do as much as possible with the data it has access to, while the YQ2 codebase tends to put a lot of the checks at the top of the function, so for consistency I tried to keep it that way.

What do you feel should happen if other->takedamage && !plane? Call T_Damage() with vec3_origin as plane?

Out of curiosity though, what bug are you referring to? Is there a execution path where a touch function gets called with NULL plane?

[the remark this answers is not in the document]
Comment (Yamagi) [reply src/g_newweap.c]: > What do you feel should happen if other->takedamage && !plane? Call T_Damage() with vec3_origin as plane?

Either that or just return without doing anything.

> Out of curiosity though, what bug are you referring to? Is there a execution path where a touch function gets called with NULL plane?

I think that @DanielGibson found at least one place were plane might be NULL. But I don't know were.

[the remark this answers is not in the document]
Comment (BjossiAlfreds) [reply src/g_newweap.c]: I've pushed a change to blaster2_touch. Is this closer to what you have in mind?

**Comment by Yamagi** [comment]
Aaaaand merged! :)

**Comment by BjossiAlfreds** [comment]
I have updated the xatrix PVS branch with all weapon touch functions updated in the same way as here (made a handy little wrapper function for the copy). I also went through other files to make sure other entities touch functions are correct.

Would you like me to do a 2nd PR for rogue that addresses the rest of the touch functions?

**Comment by Yamagi** [comment]
Thank you for doing xatrix, that PR is the next one on the review list. Another pull request for rogue would be nice. If i had to do it myself it would properly take days or even weeks...

**Pull request merged**